kolibri/plugins/coach/assets/src/views/reports/ReportsLessonExerciseLearnerListPage.vue

```diff
@@ -73,26 +60,78 @@
 
   import commonCoach from '../common';
   import { PageNames } from '../../constants';
+  import { LastPages } from '../../constants/lastPagesConstants';
   import ReportsLessonExerciseHeader from './ReportsLessonExerciseHeader';
+  import ReportsExerciseLearners from './ReportsExerciseLearners';
 
   export default {
     name: 'ReportsLessonExerciseLearnerListPage',
     components: {
       ReportsLessonExerciseHeader,
+      ReportsExerciseLearners,
     },
     mixins: [commonCoach],
+    data() {
+      return {
+        viewByGroups: Boolean(this.$route.query && this.$route.query.groups),
+      };
+    },
     computed: {
       lesson() {
         return this.lessonMap[this.$route.params.lessonId];
       },
-      recipients() {
-        return this.getLearnersForGroups(this.lesson.groups);
+      summaryTally() {
+        const recipients = this.getLearnersForGroups(this.lesson.groups);
+        return this.getContentStatusTally(this.$route.params.exerciseId, recipients);
+      },
+      nonEmptyGroups() {
+        if (!this.groups || !this.groups.length) {
```

Comment:
We should only show tables for groups that have been assigned the Lesson. `this.groups` will return all of the groups in the classroom.

Reply:
Oh didn't notice, thanks, good catch. Will check it.

Reply:
Done. I also noticed that all groups were displayed in a table in Groups column so I filtered them by lesson groups, too. Otherwise it would be misleading when switching from groups to the list of all learners.